### ISSUE TITLE
Create icon component with utilities and style guide

### ISSUE THREAD

**Issue by cnotv** [rancher/dashboard#7588]
# Describe 

Given 142 occurrences, utilities and [existing style guide](https://rancher.github.io/storybook/?path=/story/foundation-icons--page), create a component for the icon matching existing criteria and utilities.

Considered cases:
- size
- spacing
- aria/a11y

## Context

It is currently available a [list of icons used for Rancher](https://rancher.github.io/icons/), although we do not display the used sizes within the Dashboard and potentially globally.
List of utilities can be found [here](https://github.com/rancher/dashboard/blob/master/shell/assets/styles/fonts/_icons.scss) and range from size to animation and hierarchy.

This lack has been highlighted by issues encountered while dealing with the icons in https://github.com/rancher/dashboard/issues/7496

While creating this component, address existing ones: IconIsDefault (it seems not used but listed in [Storybook](https://rancher.github.io/storybook/?path=/docs/form-iconisdefault--default-story)), and IconText (which also seems not defined).

<img width="247" alt="Screenshot 2023-10-04 at 09 44 09" src="https://github.com/rancher/dashboard/assets/5009481/83dd3295-4345-4ad9-8ea9-dcd3192110cc">

**Comment by gaktive** [comment]
Pushing out some Storybook settings out from Q2 to Q3 though as we get closer to the Q2 release and there's time left, we could have EMEA pull in some back in.

**Comment by cnotv** [comment]
@rak-phillip should we integrate this to the `@rancher/components` library just as for the buttons?

**Comment by rak-phillip** [comment]
@cnotv yeah, I think an icon component would be a great candidate. Rancher Desktop suffers from the same problem described above.

**Comment by codyrancher** [comment]
See https://github.com/rancher/dashboard/issues/9853#issuecomment-3356299420
